### ISSUE TITLE
[Bug]: Error doing job: Fatal error: protocol.data_received() call failed.

### ISSUE THREAD

**Issue by ljw092** [custom-components/ble_monitor#870]
### What happened?

Sometimes it will report this error
ERROR (Thread-3) [homeassistant] Error doing job: Fatal error: protocol.data_received() call failed.

```
2022-05-23 01:19:31 ERROR (Thread-3) [homeassistant] Error doing job: Fatal error: protocol.data_received() call failed.
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/asyncio/selector_events.py", line 870, in _read_ready__data_received
    self._protocol.data_received(data)
  File "/usr/local/lib/python3.9/site-packages/aioblescan/aioblescan.py", line 1855, in data_received
    self.process(packet)
  File "/config/custom_components/ble_monitor/__init__.py", line 662, in process_hci_events
    sensor_msg, tracker_msg = self.ble_parser.parse_data(data)
  File "/config/custom_components/ble_monitor/ble_parser/__init__.py", line 170, in parse_data
    sensor_data = parse_xiaomi(self, service_data, mac, rssi)
  File "/config/custom_components/ble_monitor/ble_parser/xiaomi.py", line 843, in parse_xiaomi
    result.update(resfunc(dobject))
  File "/config/custom_components/ble_monitor/ble_parser/xiaomi.py", line 212, in obj0010
    return {'toothbrush': 0, 'score': xobj[1]}
IndexError: index out of range
```

```
2022-05-22 23:53:33 ERROR (Thread-3) [homeassistant] Error doing job: Fatal error: protocol.data_received() call failed.
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/asyncio/selector_events.py", line 870, in _read_ready__data_received
    self._protocol.data_received(data)
  File "/usr/local/lib/python3.9/site-packages/aioblescan/aioblescan.py", line 1855, in data_received
    self.process(packet)
  File "/config/custom_components/ble_monitor/__init__.py", line 662, in process_hci_events
    sensor_msg, tracker_msg = self.ble_parser.parse_data(data)
  File "/config/custom_components/ble_monitor/ble_parser/__init__.py", line 118, in parse_data
    shortened_local_name = adstruct[2:].decode("utf-8")
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xff in position 1: invalid start byte
```

### Sensor type

I have several Xiaomi sensors (LYWSD02, LYWSDCGQ, LYWSD03MMC), I don't know which one is reporting the error.

**Comment by Ernst79** [comment]
Fix is in #871 for both errors. I will do some test later today, before releasing it.

**Comment by Ernst79** [comment]
It looks like there is something wrong with the bluetooth advertisements you receive. The last message shows you that you have received a message of the type "manufacturer specific data". This is in the form `0x04FF112233`. `04` is the length and can be anything, `FF` is saying that you are receiving manufacturer specific data, and this is followed by a company id (`1122`) and the data (`33`). The error is saying that you have received something like `0x01FF` or `0x02FF01`, so it is missing the company id. My guess is that something is wrong with the Bluetooth receiver, 

You previous errors show similar behavior, it just looks like parts of the advertisements are missing. Can you try with a different Bluetooth dongle?

**Comment by ljw092** [comment]
These errors are actually accidental, but every time the error is reported, scan will stop, and some advertisements will not be received. Especially the door and window sensors, missing advertisements the automation cannot be triggered.

My bluetooth is onboard on Raspberry Pi 3b and used in docker. Is there something wrong with my bluetooth?

I got another error [#827](https://github.com/custom-components/ble_monitor/issues/827#issuecomment-1134181298). is this also my bluetooth problem?

**Comment by Ernst79** [comment]
All issues seem to indicate that there is some issue with your Bluetooth installation. It is hard to say what the cause is, as it only happens occasionally. The easiest is to try an external dongle, like the Asus BT400 or similar. Can be found for 5€. I can't give you guarantee for success, as it might also be something related to your installation, but my first bet would be on getting a new dongle.

**Comment by ljw092** [comment]
I found the problem, because this problem was reported in dmesg:
`Bluetooth: hci0: Frame reassembly failed (-84)`.
And I found the fix [here](https://github.com/raspberrypi/firmware/issues/1150).

But now it occasionally reports another problem in dmesg
`Bluetooth: hci0: advertising data len corrected`
But there is no problem in ha. I guess it probably doesn't matter.

Sorry for your time.
Hope it will help someone to save lots of time.
